frontend/src/app/modules/lessons/lesson/lesson.component.ts

```diff
@@ -1,27 +1,34 @@
 import { Component, Input } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
+import { BaseComponent } from '@core/base/base.component';
 import { SpinnerService } from '@core/services/spinner.service';
+import { UserService } from '@core/services/user.service';
 import { YoutubePlayerComponent } from '@shared/components/youtube-player/youtube-player.component';
 
 import { Lesson } from 'src/app/models/lessons/lesson';
 import { Question } from 'src/app/models/lessons/question';
 import { LessonsService } from 'src/app/services/lessons.service';
+import { NotificationService } from 'src/app/services/notification.service';
 
 @Component({
     selector: 'app-lesson',
     templateUrl: './lesson.component.html',
     styleUrls: ['./lesson.component.sass'],
 })
-export class LessonComponent {
+export class LessonComponent extends BaseComponent {
     @Input() lesson: Lesson;
 
     questions: Question[] = [];
 
     constructor(
         private dialogRef: MatDialog,
         private lessonsService: LessonsService,
+        private userService: UserService,
         private spinner: SpinnerService,
-    ) {}
+        private notificationService: NotificationService,
+    ) {
+        super();
```

Comment:
I wanted to inherit Tatyana's component `src/app/core/base/base.component.ts` because it clears subscriptions. Webstorm suggested it. Actually I'm not sure about it..

Reply:
ok

Reply:
I've tried but invain. By the way in our app all classes that inherit from `BaseComponent` and have a constuctor call `super()`. I believe it's necessary.